crates/router/src/utils/verification.rs

```diff
@@ -0,0 +1,112 @@
+use actix_web::web;
+#[cfg(all(feature = "olap", feature = "kms"))]
+use api_models::verifications::{self, ApplepayMerchantResponse};
+use error_stack::{Report, ResultExt};
+#[cfg(feature = "kms")]
+use external_services::kms;
+
+use crate::{
+    core::errors::{self, api_error_response},
+    headers, logger,
+    routes::AppState,
+    services, types, utils,
+};
+
+const APPLEPAY_MERCHANT_VERIFICATION_URL: &str =
+    "https://apple-pay-gateway.apple.com/paymentservices/registerMerchant";
+const APPLEPAY_INTERNAL_MERCHANT_NAME: &str = "Applepay_merchant";
+
+pub async fn verify_merchant_creds_for_applepay(
+    state: &AppState,
+    _req: &actix_web::HttpRequest,
+    body: web::Json<verifications::ApplepayMerchantVerificationRequest>,
+    kms_config: &kms::KmsConfig,
+) -> common_utils::errors::CustomResult<
+    services::ApplicationResponse<ApplepayMerchantResponse>,
+    api_error_response::ApiErrorResponse,
+> {
+    let encrypted_merchant_identifier = &state
+        .conf
+        .applepay_merchant_configs
+        .common_merchant_identifier;
+    let encrypted_cert = &state.conf.applepay_merchant_configs.merchant_cert;
+    let encrypted_key = &state.conf.applepay_merchant_configs.merchant_cert_key;
+
+    let applepay_internal_merchant_identifier = kms::get_kms_client(kms_config)
+        .await
+        .decrypt(encrypted_merchant_identifier)
+        .await
+        .change_context(api_error_response::ApiErrorResponse::InternalServerError)?;
+
+    let cert_data = kms::get_kms_client(kms_config)
+        .await
+        .decrypt(encrypted_cert)
+        .await
+        .change_context(api_error_response::ApiErrorResponse::InternalServerError)?;
+
+    let key_data = kms::get_kms_client(kms_config)
+        .await
+        .decrypt(encrypted_key)
+        .await
+        .change_context(api_error_response::ApiErrorResponse::InternalServerError)?;
```

Comment:
Can we have this decrypted similar to all the other KMS decryptions that are being done? in `kms.rs`